test/widgets/message_list_test.dart

```diff
@@ -51,6 +59,58 @@ Future<void> setupMessageListPage(WidgetTester tester, {
 void main() {
   TestZulipBinding.ensureInitialized();
 
+  group('MessageWithSender', () {
+    testWidgets('Updates avatar on RealmUserUpdateEvent', (tester) async {
+      addTearDown(TestZulipBinding.instance.reset);
+
+      RealmContentNetworkImage? findAvatarImageWidget(WidgetTester tester) {
+        final firstMessageWithSender = tester.widgetList(find.byType(MessageWithSender)).first;
+        return tester.widgetList<RealmContentNetworkImage>(
+          find.descendant(
+            of: find.byWidget(firstMessageWithSender),
+            matching: find.byType(RealmContentNetworkImage)),
+        ).singleOrNull;
+      }
+
+      void checkResultForSender(User sender) {
+        final avatarUrl = sender.avatarUrl;
+        switch (avatarUrl) {
+          case String(): {
+            check(findAvatarImageWidget(tester)).isNotNull().src.equals(resolveUrl(avatarUrl, eg.selfAccount));
+          }
+          case null: {
+            check(findAvatarImageWidget(tester)).isNull();
+          }
+        }
+      }
+
+      Future<void> handleNewAvatarEventAndPump(WidgetTester tester, String avatarUrl) async {
+        final store = await TestZulipBinding.instance.globalStore.perAccount(eg.selfAccount.id);
+        store.handleEvent(RealmUserUpdateEvent(id: 1, userId: eg.selfUser.userId, avatarUrl: avatarUrl));
+        await tester.pump();
+      }
+
+      final httpClient = FakeHttpClient();
+      debugNetworkImageHttpClientProvider = () => httpClient;
+      httpClient.request.response
+        ..statusCode = HttpStatus.ok
+        ..content = kSolidBlueAvatar;
+
+      await setupMessageListPage(tester, narrow: const AllMessagesNarrow());
+      checkResultForSender(eg.selfUser);
+
+      await handleNewAvatarEventAndPump(tester, '/foo.png');
+      // TODO only vary [avatarUrl], not other fields
+      checkResultForSender(eg.user(userId: eg.selfUser.userId, avatarUrl: '/foo.png'));
```

Comment:
This `checkResultForSender` is a private helper for this test, right? And it only consults the `avatarUrl`.

So it seems like you could resolve this TODO by just passing the `avatarUrl` value itself.